src/Helper/HeadMeta.php

```diff
@@ -458,20 +464,17 @@ public function set($value)
      * Not valid in a non-HTML5 doctype
      *
      * @param  string $charset
-     * @throws Exception\RuntimeException
+     * @param  Exception\RuntimeException
      * @return HeadMeta Provides a fluent interface
      */
     public function setCharset($charset)
     {
-        if ($this->view->plugin('doctype')->isXhtml()) {
-            throw new Exception\RuntimeException('XHTML* doctype has no attribute charset');
-        }
-
         $item = new stdClass;
         $item->type = 'charset';
         $item->charset = $charset;
         $item->content = null;
         $item->modifiers = [];
+        $this->isValid($item);
```

Comment:
If false, then throw `Exception\InvalidArgumentException` here.

Reply:
Compare with the methods `set`, `prepend` etc.

Reply:
updated